### PR TITLE
docs:Add documentation for TLS configuration parameters for HTTP and gRPC

### PR DESCRIPTION
**Pull request by andretadeu**
**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:
Improve documentation regarding TLS.

**Special notes for your reviewer**:

**Checklist**
- [X] Documentation improved

### CODE REVIEW

File: docs/sources/configuration/_index.md [hunk 1]
```diff
@@ -196,6 +196,28 @@ configures the HTTP and gRPC server communication of the launched service(s).
...
+  [key_file: <string> | default = ""]
+  # HTTP TLS Client Auth type.
+  # CLI flag: -server.http-tls-client-auth
+  # Possible values:
+  # - RequestClientCert
+  # - RequireAnyClientCert
+  # - RequireClientCert
+  # - VerifyClientCertIfGiven
+  # - RequireAndVerifyClientCert
+  # - "" (Empty string) or "NoClientCert"
```
Comment (ssncferreira) [inline docs/sources/configuration/_index.md:215]: No need to add these possible values. As this is related to the weaveworks library 
```suggestion
```

**Review by ssncferreira — COMMENTED** [review]
LGTM. This looks great, thank you for the contribution 🎉 
Just a small comment

File: docs/sources/configuration/_index.md (hunk 1 above)
Comment (andretadeu) [reply docs/sources/configuration/_index.md]: I removed the suggested lines in a new commit.

File: docs/sources/configuration/_index.md [hunk 2]
```diff
@@ -204,6 +226,28 @@ configures the HTTP and gRPC server communication of the launched service(s).
...
+  [key_file: <string> | default = ""]
+  # gRPC TLS Client Auth type.
+  # CLI flag: -server.grpc-tls-client-auth
+  # Possible values:
+  # - RequestClientCert
+  # - RequireAnyClientCert
+  # - RequireClientCert
+  # - VerifyClientCertIfGiven
+  # - RequireAndVerifyClientCert
+  # - "" (Empty string) or "NoClientCert"
```
[the remark this answers is not in the document]
Comment (andretadeu) [reply docs/sources/configuration/_index.md]: Same here.

**Review by ssncferreira — APPROVED** [review]
LGTM 🎉 thank you for addressing the comments 🙂

**Pull request merged**